src/subscriptions-register.jsx

```diff
@@ -75,19 +75,10 @@ class SubscriptionRegisterDialog extends React.Component {
             );
         }
         let insights;
-        let insights_checkbox_disabled = true;
-        if (this.props.insights_available === true) {
-            insights_checkbox_disabled = false;
-        } else {
-            if (this.props.auto_attach === true) {
-                insights_checkbox_disabled = false;
-            }
-        }
```

Comment:
Part of this deleted code is still needed though: in particular, the insights checkbox must be disabled when insights-client is not available; the only needed part that can be removed is the `else {...}` block.

Reply:
When insights-client is not available, then it will be automatically installed. When content is always enabled, then it could not happen that it will not be possible to install insights-client during registration process. Thus, this change is correct.

Reply:
You are wrong. Function `detectInsights` tries to detect following things. First, it checks if `insights-client` command is possible to execute. If `insights-client` command does not exists, then it checks if PackageKit is available. It really does not check if `insights-client` RPM is available in any repository.

Reply:
My original point still stands: the code there disables the possibility to register to Insights if `insights-client` is not installed, and PackageKit is available. While it is a corner case, it needs to still be kept into account.

Reply:
Removed, or even only disabled by the user. That's why I said "available" previously: "not installed" and "installed and disabled" are basically the same, from our POV.